src/Orchard.Web/Modules/Orchard.OutputCache/Filters/OutputCacheFilter.cs

```diff
@@ -588,6 +608,44 @@ public class OutputCacheFilter : FilterProvider, IActionFilter, IResultFilter, I
             }
         }
 
+        private byte[] ReplaceRequestVerificationTokenWithBeaconTag(byte[] source, Encoding encoding) {
+            // Because of the __RequestVerificationToken hidden field vary by the user, before caching the output, we need to replace the 
+            // __RequestVerificationToken hidden field with a "beacon" text that will be replaced before rendering the page
+            // with a fresh new __RequestVerificationToken hidden field.
+            // What we do is to replace every <input name="__RequestVerificationToken " value="{the-value}" /> with
+            //                                <$request-verification-token-beacon-for-hidden-field />
+            if (PreventCachingRequestVerificationToken()) {
+                string patternRegion = "<\\s*input\\s*.*name\\s*=\\s*\"__RequestVerificationToken\"\\s*.*value\\s*=\\s*\"" +
+                                       "(?<value>[\\w-]*)" +
+                                       "\"\\s*/>";
+                RegexOptions regexOptions = RegexOptions.Multiline | RegexOptions.IgnoreCase | RegexOptions.Compiled;
```

Comment:
Try not to use a regex, it's dangerous. Or at least define a small timeout value. Also before using the regex, so a simple IndexOf to ensure there is a potential match first.

Reply:
The new commit adds indexof check and a regex timeout as suggested. I modified the regex replace with a static one in order to be more performant. 
In our dev environment:
With the regex object it performs a replace in 20.68 millisenconds
With the static regex it performs a replace in 4.23 millisenconds

Reply:
But do you really need a regex? Why not simply search for `<input name="__RequestVerificationToken"` since it's generated by some well-known code. ? If you search for spaces between `name` and `=` why not seach for other attributes before `name`? Because you are making an assumption about what is generating this input, and the same assumption lets you search for the exact string I suggested, removing the need of a regex.

Reply:
I removed regex as suggested, see latest commit.